src/LiveComponent/src/Hydration/ArrayCollectionHydrationExtension.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Hydration;
+
+use Doctrine\Common\Collections\ArrayCollection;
+
+/**
+ * @author Jean-Paul van der Wegen <info@jpvdw.nl>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+class ArrayCollectionHydrationExtension extends AbstractDoctrineHydrationExtension implements HydrationExtensionInterface
+{
+    public function supports(string $className): bool
+    {
+        return ArrayCollection::class === $className;
+    }
+
+    public function hydrate(mixed $value, string $className): ?object
+    {
+        $output = new ArrayCollection();
+        foreach ($value as $item) {
+            $object = $this->findObject($item['class'], $item['identifierValue']);
```

Comment:
No control on the $item keys ?

Reply:
In the current setup there is not. Off-course the (de)hydration process will be in the same order. 

I believe that for this use case this approach should be enough. But adding index key is not that big of change. You preference to do this?

Reply:
Sorry, what i meant was: what if "class" or "identifierValue" is missing from $item ?

Reply:
just add some checks and throw exception when is not present.